### PR TITLE
Removes jitteriness - Inline Parameter Name Hints

### PR DESCRIPTION
**Pull request by akhera99**
Fixes: https://github.com/dotnet/roslyn/issues/46610

Hello,

Jason and I found that the DesiredSize property on the Border was getting changed by a small amount when changing the editor. We found this comment:
```
// Note: Some adornments may change size when added to the view's visual tree due to inherited
// dependency properties that affect layout. Such options can include SnapsToDevicePixels,
// UseLayoutRounding, TextRenderingMode, TextHintingMode, and TextFormattingMode. Making sure
// that these properties on the adornment match the view's values before calling Measure here
// can help avoid the size change and the resulting unnecessary re-format.
```

on the IntraTextAdornmentTagger.cs which basically states that certain properties need to be set for the UI element in order for them to not get resized. So we made sure to set those properties using the view's values before calling Measure. This required a shift from using a TextView to a WpfTextView, which was just casted in the InlineParameterNameHintsTagger.cs

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs [hunk 1]
```diff
@@ -18,18 +21,22 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
...
-            : base(CreateElement(text, lineHeight, format), removalCallback: null, PositionAffinity.Predecessor)
+        public InlineParameterNameHintsTag(string text, IWpfTextView textView, TextFormattingRunProperties format)
+            : base(CreateElement(text, textView, format), removalCallback: AdornmentCallbackFunction, PositionAffinity.Predecessor)
+        {
+        }
+
+        private static void AdornmentCallbackFunction(object tag, UIElement element)
         {
+            //
         }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs:37]: what is this empty method for?

File: src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs [hunk 2]
```diff
@@ -18,18 +21,22 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
     internal class InlineParameterNameHintsTag : IntraTextAdornmentTag
     {
         public const string TagId = "inline parameter name hints";
-
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs:21]: add back plz :)

File: src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs:37]: Oops this was some debugging code that got left in.

File: src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs [hunk 3]
```diff
@@ -52,13 +59,21 @@ private static UIElement CreateElement(string text, double lineHeight, TextForma
...
+                Margin = new Thickness(0, -0.20 * textView.LineHeight, 5, 0),
                 Padding = new Thickness(1),
-                VerticalAlignment = VerticalAlignment.Center,
+                SnapsToDevicePixels = textView.VisualElement.SnapsToDevicePixels,
+                UseLayoutRounding = textView.VisualElement.UseLayoutRounding,
+                VerticalAlignment = VerticalAlignment.Center
             };
 
+            // Need to set these properties to avoid unecessary reformatting because some dependancy properties
+            // affect layout
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs:72]: Add the comment to SnapsToDevicePixels and UseLayoutRounding as well?

File: src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs (hunk 1 above)
Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs:37]: Sorry, forgot to remove it

File: src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs (hunk 3 above)
Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs:72]: Will add there as well

**Comment by davidwengier** [comment]
Thank you for this, I'm looking forward to re-upping my caffeine intake once the jitters are gone 😁

**Comment by jasonmalinowski** [comment]
I retargeted this to 16.8 Preview 2 as this is by far the biggest source of feedback for this new feature -- OK @jinujoseph and @vatsalyaagrawal to take that in there?

**Pull request merged**